iree/turbine/kernel/wave/compile.py

```diff
@@ -33,6 +33,7 @@ def __init__(
         asm: str,
         gpu_binary_path: Optional[str],
         bound_scalar_symbols: dict[IndexSymbol, int],
+        symbols_args_map: dict[IndexSymbol, tuple[int, int]],
```

Comment:
is the type dict[IndexSymbol, tuple[int, int | None]], since the shape can be dynamic? Also if this is not optional, should it be before gpu_binary_path?